src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -296,103 +297,108 @@ def get_top_5_issues_by_count_for_file(
-        query=(
-            Query(subquery)
-            .set_select(
-                [
-                    Column("function_name"),
-                    Function(
-                        "arrayElement",
-                        (Function("groupArray", [Column("group_id")]), 1),
-                        "group_id",
-                    ),
-                    Function(
-                        "arrayElement",
-                        (Function("groupArray", [Column("event_count")]), 1),
-                        "event_count",
-                    ),
-                ]
+                        ),
+                        Condition(Function("notHandled", []), Op.EQ, 1),
+                    ]
+                )
+                .set_orderby([OrderBy(Column("event_count"), Direction.DESC)])
             )
-            .set_groupby(
-                [
-                    Column("title"),
-                    Column("culprit"),
-                    Column("function_name"),
-                ]
+
+            # filter on the subquery to squash group_ids with the same title and culprit
+            # return the group_id with the greatest count of events
+            request = SnubaRequest(
+                dataset=Dataset.Events.value,
+                app_id="default",
+                tenant_ids={"organization_id": projects[0].organization_id},
+                query=(
+                    Query(subquery)
+                    .set_select(
+                        [
+                            Column("function_name"),
+                            Function(
+                                "arrayElement",
+                                (Function("groupArray", [Column("group_id")]), 1),
+                                "group_id",
+                            ),
+                            Function(
+                                "arrayElement",
+                                (Function("groupArray", [Column("event_count")]), 1),
+                                "event_count",
+                            ),
+                        ]
+                    )
+                    .set_groupby(
+                        [
+                            Column("title"),
+                            Column("culprit"),
+                            Column("function_name"),
+                        ]
+                    )
+                    .set_orderby([OrderBy(Column("event_count"), Direction.DESC)])
+                    .set_limit(5)
+                ),
             )
-            .set_orderby([OrderBy(Column("event_count"), Direction.DESC)])
-            .set_limit(5)
-        ),
-    )
-    return raw_snql_query(request, referrer=Referrer.GITHUB_PR_COMMENT_BOT.value)["data"]
+            return raw_snql_query(request, referrer=Referrer.GITHUB_PR_COMMENT_BOT.value)["data"]
+        except Exception as e:
+            logger.exception("github.open_pr_comment.snuba_query_error", extra={"error": str(e)})
```

Comment:
maybe we should do `sentry_sdk.capture_exception` here? iirc it'll raise a handled error. it's harder to monitor loggers

Reply:
Sentry will automatically capture logger exceptions as we use the logging integration for python. no need to log the stacktrace along with it.
https://docs.sentry.io/platforms/python/integrations/logging/#verify

Reply:
additionally, i do not think this makes sense to be a sentry error -- IMO this should just be a metric you have on a dashboard and can set thresholds. 

Unless you are unsure why it is erroring.

Reply:
we are unsure why it is erroring

Reply:
usually it errors because the query itself is too big or too much memory is allocated in snuba to process the query

Reply:
I also am not sure if you want to treat the case where you try again differently vs failing it the first time.

I think you might, as they will result in different outcomes, correct? (i.e. second failure = feature not working at all, first failure = feature working, but degraded)

Reply:
> usually it errors because the query itself is too big or too much memory is allocated in snuba to process the query

Shouldn't there already be errors in sentry/snuba for this then?

Reply:
yes, we would like to make the query smaller so it does not do any of those things